contrib/runners/mistral_v2/mistral_v2.py

```diff
@@ -363,6 +363,80 @@ def resume(self, ex_ref, task_specs):
         wait_exponential_multiplier=cfg.CONF.mistral.retry_exp_msec,
         wait_exponential_max=cfg.CONF.mistral.retry_exp_max_msec,
         stop_max_delay=cfg.CONF.mistral.retry_stop_max_msec)
+    def pause(self):
+        mistral_ctx = self.context.get('mistral', dict())
+
+        if not mistral_ctx.get('execution_id'):
+            raise Exception('Unable to pause because mistral execution_id is missing.')
+
+        # Pause the main workflow execution. Any non-workflow tasks that are still
+        # running will be allowed to complete gracefully.
+        self._client.executions.update(mistral_ctx.get('execution_id'), 'PAUSED')
+
+        # If workflow is executed under another parent workflow, pause the corresponding
+        # action execution for the task in the parent workflow.
+        if 'parent' in getattr(self, 'context', {}):
+            mistral_action_ex_id = mistral_ctx.get('action_execution_id')
+            self._client.action_executions.update(mistral_action_ex_id, 'PAUSED')
+
+        # Identify the list of action executions that are workflows and cascade pause.
+        for child_exec_id in self.execution.children:
+            child_exec = ActionExecution.get(id=child_exec_id, raise_exception=True)
+            if (child_exec.runner['name'] in action_constants.WORKFLOW_RUNNER_TYPES and
```

Comment:
Shouldn't we check here for mistral-v2 runner type since action-chain is also in `WORKFLOW_RUNNER_TYPES`.

I know it should also work this way because we already filter on execution id, but still.